frontend/src/static/js/v_authorship.js

```diff
@@ -245,20 +246,29 @@ window.vAuthorship = {
             ? this.getContributionFromAllAuthors(contributionMap)
             : contributionMap[this.info.author];
 
-        if (lineCnt) {
+        // skip file not touched by author
+        if (!this.info.isMergeGroup && !(this.info.author in contributionMap)) {
+          return;
+        }
```

Comment:
what does this section do with the binary file change?

Reply:
This is contributed by the original author. It skips the files not touched by the current author. I am not sure but removing this section seem to cause bug where loading of authorship cannot terminate. I am looking into the problem but it may be necessary to keep it as it is at this moment.

Reply:
Yes, but the line here looks very similar to the previous section. Instead of conditioning on linecount and isBinary, can we have a function like isValidFile? Also, you can calculate linecount after the if condition.

Reply:
Sure, I will refactor this section.